dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -151,6 +151,85 @@ void dpnp_tril_c(void* array_in, void* result1, const int k, size_t* shape, size
     return;
 }
 
+template <typename _DataType>
+void dpnp_triu_c(void* array_in, void* result1, const int k, size_t* shape, size_t* res_shape, const size_t ndim, const size_t res_ndim)
+{
+    _DataType* array_m = reinterpret_cast<_DataType*>(array_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    size_t res_size = 1;
+    for (size_t i = 0; i < res_ndim; ++i)
+    {
+        res_size *= res_shape[i];
```

Comment:
It doesn't work correct when shape element is zero. Do checks before multiply.
```python
>>> a = numpy.ones(shape=(1, 2, 0))
>>> a.shape
(1, 2, 0)
```

And add test case for testing with this kind given shape.